contrib/enfuseAdvanced.lua

```diff
@@ -612,434 +210,851 @@ local function GetFileName(full_path)
 	local extension = string.match(full_path, "%.%w*")
     return path, filename, identifier, extension
 end
-local function truncate(x)
-      return x<0 and math.ceil(x) or math.floor(x)
-end 
-local function replace_comma_to_dot(s)
-	return string.gsub(s, "%,", ".")
-end
-local function remove_temp_files()
-	if dt.configuration.running_os == "windows" then
-		dt.print_log("enfuseAdvanced - Deleting temp files...")
-		dt.control.execute("del "..images_to_align)
-		dt.control.execute("del "..images_to_blend)
-		dt.print_log("enfuseAdvanced - Done deleting")
-	else
-		dt.print_log("enfuseAdvanced - Deleting temp files...")
-		dt.control.execute("rm "..images_to_align)
-		dt.control.execute("rm "..images_to_blend)
-		dt.print_log("enfuseAdvanced - Done deleting")
-	end
+
+local function CleanSpaces(text) --removes spaces from the front and back of passed in text
+	text = string.gsub(text,'^%s*','')
+	text = string.gsub(text,'%s*$','')
+	return text
 end
-local function build_execute_command(cmd, args, file_list)
-	local result = cmd.." "..args.." "..file_list
+
+local function BuildExecuteCmd(prog_table) --creates a program command using elements of the passed in program table
+	local result = CleanSpaces(prog_table.bin)..' '..CleanSpaces(prog_table.arg_string)..' '..CleanSpaces(prog_table.images_string)
 	return result
 end
-local function copy_exif(from_file, to_file)
-	exifStartCommand = exiftool_path.." -TagsFromFile "..df.sanitize_filename(from_file).." -exif:all --subifd:all -overwrite_original "..df.sanitize_filename(to_file)
-	dt.print_log("enfuseAdvanced - EXIFTool Start Command: "..exifStartCommand)
-	resultexif=dsys.external_command(exifStartCommand)
-	if (resultexif == 0) then
-		dt.print_log("enfuseAdvanced - EXIFTool copy successful")
-		dt.print("Copied EXIF data")
-	else
-		dt.print(_('ERROR: exiftool doesn\'t work. for more informations see terminal output'))
-		dt.print_error("exif copy failed")
+
+local function PreCall(prog_tbls) --looks to see if this is the first call, if so checks to see if program is installed properly
+	for _,prog in pairs(prog_tbls) do
+		if prog.first_run then
+			prog.bin = df.check_if_bin_exists(prog.name)
+			if not prog.bin then 
+				prog.install_error = true
+			else
+				prog.bin = CleanSpaces(prog.bin)
+			end
```

Comment:
Have you had a problem with executable paths having spaces?  They are stored in the darktablerc file without leading or trailing space, so I don't think you need this.

Reply:
No I don't think I have. I did this just as a double check. The CleanSpaces() function is needed elsewhere, so since I already had it I figured I'd just be safe and run the binaries through it too.